docs: add warning on Validation $data array (#7515)
codeigniter4/CodeIgniter4: user_guide_src/source/libraries/validation.rst

Allowing Parameters
-------------------

If your method needs to work with parameters, the function will need a minimum of three parameters:

1. the value to validate (``$value``)
2. the parameter string (``$params``)
3. an array with all of the data that was submitted the form (``$data``)
4. (optional) a custom error string (``&$error``), just as described above.

> .. warning:: The other field values in ``$data`` is unvalidated (or maybe invalid)
>     data. Using unvalidated input data is a source of vulnerability. You must

Reviewer: ```suggestion
.. warning:: The field values in ``$data`` are unvalidated (or may be invalid).
    Using unvalidated input data is a source of vulnerability. You must
```
